pymatgen/entries/compatibility.py

```diff
@@ -991,45 +827,247 @@ def __init__(
         )
 
 
-class MaterialsProject2020Compatibility(CorrectionsList):
+class MaterialsProject2020Compatibility(Compatibility):
     """
     This class implements the GGA/GGA+U mixing scheme, which allows mixing of
     entries. Note that this should only be used for VASP calculations using the
-    MaterialsProject parameters (see pymatgen.io.vaspio_set.MPVaspInputSet).
+    Materials Project input set parameters (see pymatgen.io.vasp.sets.MPRelaxSet).
     Using this compatibility scheme on runs with different parameters is not
     valid.
     """
 
     def __init__(
-        self, compat_type="Advanced", correct_peroxide=True, check_potcar_hash=False
+        self,
+        compat_type="Advanced",
+        correct_peroxide=True,
+        check_potcar_hash=False,
+        input_set=MPRelaxSet,
```

Comment:
@mkhorton @shyuep can you please advise on how to resolve a serialization problem I'm having? I want to pass `MPRelaxSet` as a kwarg to this class so that it can be used by `PotcarCorrection`, which only needs to access the class attribute `CONFIG`. I don't want to instantiate the class because that requires an input structure, which wouldn't make sense to provide here. However, if the class isn't instantiated, it won't serialize, and I fail the `as_dict()` test.

Reply:
There are a number of different ways to do this... you could pass the kwarg as a string and access the class on demand, for example (e.g. `{subcls.__name__: subcls for subcls in DictSet.__subclasses__()}`). The larger question is if you should do it at all, I guess. The `CONFIG` attribute itself isn't enforced by `DictSet`, so in principle it's possible to write a input set for which you couldn't use the `PotcarCorrection`. Not really a problem in practice but feels a little messy.

Reply:
Thanks, I'm leaning towards "we should not." The MP2020 compatibility class is fitted to and therefore inextricably linked to MPRelaxSet calculations, so it probably doesn't make sense to ever check the POTCARs against any other `InputSet`. If I hard-code `MPRelaxSet` into the class instead of making it an attribute, this serialization problem will go away. Thoughts?